### DISCUSSION TITLE
Suggestion:  Include the Feathers example app in the mono-repo

### DISCUSSION THREAD

**Discussion by robbyphillips** [feathersjs/feathers#2233] [Ideas]
This came up in a slack thread, but I thought it would be worth elaborating here.

I propose adding a new top-level directory to the mono-repo called `examples/` or something to that effect.  This directory could house the official feathers sample app, and possibly more.

The major benefit for moving the app into the main repo is that we could use it as a canary for breaking changes -- especially typing only changes, or detecting when @types/express breaks something.

The only downside that I'm aware of is that trying out the sample app would be less convenient, since it would require cloning the whole mono-repo and `cd`'ing to the example directory.

Thoughts?

Edit to add a visual:
```
.
├── (mono-repo config files)
├── packages/
│   ├── adapter-commons
│   ├── authentication
│   ├── errors
│   ├── feathers
│   └── ... etc
└── examples/
    ├── feathers-chat-ts
    └── feathers-chat (?)
```

**Comment by robbyphillips** [comment]
We could also give instructions to use [degit](https://github.com/Rich-Harris/degit) to get to the example, rather than cloning the whole repo.

```
npx degit feathersjs/feathers/examples/sample-app
```

Still not as familiar as a simple `git clone`, but should be a pretty low barrier.

**Comment by tvld** [comment]
For sure, I would love the idea for as many comprehensive examples as possible. But I rather would like to see it as an effort of the community to expand on the documentation.
Personally I am not a big fan of extending the core repo, as everything need to be carefully checked for security purposes. The less code in the core the better, as for me at least...))

**Reply by robbyphillips** [reply]
The suggestion is to use the already existing [sample app](https://github.com/feathersjs/feathers-chat-ts) as a canary in the build process. 

Building the apps in the examples directory during the CI process would help catch things like breaking typescript types, and using the extant example apps would mean less new work to create examples.
```
.
├── (mono-repo config files)
├── packages/
│   ├── adapter-commons
│   ├── authentication
│   ├── errors
│   ├── feathers
│   └── ... etc
└── examples/
    ├── feathers-chat-ts
    └── feathers-chat (?)
```
